sdks/java/core/src/main/java/org/apache/beam/sdk/util/Credentials.java

```diff
@@ -132,15 +135,32 @@ public static Credential getCredential(GcpOptions options)
    * Loads OAuth2 credential from a local file.
    */
   private static Credential getCredentialFromFile(
-      String keyFile, String accountId, Collection<String> scopes)
+          @Nullable String keyFile, @Nullable String accountName, Collection<String> scopes)
       throws IOException, GeneralSecurityException {
-    GoogleCredential credential = new GoogleCredential.Builder()
+
+    if (keyFile == null) {
+      throw new IOException("If accountName is given, also supply a keyFile");
+    }
```

Comment:
Note that all these options are given via the command line. So you really want to check all the permutations. So what is valid:

p12 keyfile + accountName
json keyfile : NO accountName

the rest is invalid. (I've added all the errors in the test.

So the scenario were handling here is (you have given a accountName, but an accountName is not enough, you need a keyFile):
```
  @Test
  public void testAccountNameWithoutKeyFile() throws Exception {
    GcpOptions options = PipelineOptionsFactory.as(GcpOptions.class);
    options.setServiceAccountName("test@gcloud");

    thrown.expect(IOException.class);
    thrown.expectMessage(new StringContains("also supply a keyFile"));
    GcpCredentialFactory.fromOptions(options).getCredential();
  }
```

Hope it makes sense?!